### PR TITLE
WFLY-17445 Upgrade Infinispan to 14.0.5.Final

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-17445

N.B. com.squareup:protoparser is no longer shaded by org.infinispan.protostream:protostream

### CODE REVIEW

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/single/infinispan/query/HotRodClientTestCase.java [hunk 1]
```diff
@@ -61,49 +65,48 @@
...
-        RemoteCache<String, Object> remoteCache = remoteCacheManager.getCache();
-        remoteCache.clear();
-        return remoteCache;
-    }
-
-    private static Configuration createConfiguration() {
+    @Before
+    public void initialize() {
+        List<GeneratedSchema> schemas = ServiceLoader.load(SerializationContextInitializer.class, this.getClass().getClassLoader()).stream().map(ServiceLoader.Provider::get).filter(GeneratedSchema.class::isInstance).map(GeneratedSchema.class::cast).collect(Collectors.toList());
+System.out.println("LOADED SCHEMAS: " + schemas.stream().map(GeneratedSchema::getProtoFileName).collect(Collectors.toList()));
```
Comment (rhusar) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/single/infinispan/query/HotRodClientTestCase.java:83]: Unintended `System.out`.

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/single/infinispan/query/RemoteQueryTestCase.java [hunk 2]
```diff
@@ -166,40 +134,33 @@ public void testRemoteQuery() throws Exception {
...
 
-    private void registerSchema(RemoteCacheManager rcm, String key, String protoFile) {
-        RemoteCache<String, String> metadataCache = rcm.getCache(ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME);
-        metadataCache.put(key, protoFile);
-        assertFalse(metadataCache.containsKey(ProtobufMetadataManagerConstants.ERRORS_KEY_SUFFIX));
+        ConfigurationBuilder builder = new ConfigurationBuilder();
+        builder.addServer().host(INFINISPAN_SERVER_ADDRESS);
+        builder.marshaller(new ProtoStreamMarshaller());
+        for (GeneratedSchema schema : schemas) {
+            System.out.println("ADDING " + schema.getProtoFileName());
```
Comment (rhusar) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/single/infinispan/query/RemoteQueryTestCase.java:154]: Unintended `System.out`.

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/infinispan/protostream/main/module.xml [hunk 3]
```diff
@@ -33,9 +33,10 @@
         <module name="java.logging"/>
         <module name="java.sql"/>
 
-        <module name="javax.sql.api"/>
+        <!--module name="javax.sql.api"/-->
```
Comment (rhusar) [inline ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/infinispan/protostream/main/module.xml:36]: What's the purpose of commenting this out?

**Review by rhusar — CHANGES_REQUESTED** [review]

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/single/infinispan/query/HotRodClientTestCase.java (hunk 1 above)
Comment (pferraro) [reply testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/single/infinispan/query/HotRodClientTestCase.java:83]: Whoops!  I never pushed my final changes.  Ignore the noise for now.

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/infinispan/protostream/main/module.xml (hunk 3 above)
Comment (pferraro) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/infinispan/protostream/main/module.xml:36]: This was me testing to see if this was still necessary (it isn't).

Comment (rhusar) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/infinispan/protostream/main/module.xml]: Yeah, drop!

**Review by rhusar — APPROVED** [review]
Looks good and especially thanks for cleanup in the `org.jboss.as.test.clustering.single.infinispan` packages.

**Pull request merged**